### ISSUE TITLE
Express constraints on specific values in a list

### ISSUE THREAD

**Issue by bmaclach** [JacquesCarette/Drasil#1796]
Currently in Drasil, constraints can either be expressed as a Range, EnumeratedReal, or EnumeratedStr.

Another type of constraint we need to express are constraints on specific elements of a list. An example in SSP is that the first and last elements of the list __x__<sub>wt</sub> must be the same as the first and last elements of the list __x__<sub>slope</sub>. 

This might be more an issue of adding to `Expr` a way to access an element of a vector. It also requires being able to express constraints on equality, but related issue #1795 covers that.

**Comment by JacquesCarette** [comment]
You are correct that the right way to do this first involves fixing #1795 and then to change `Expr` to do indexing properly.

**Comment by JacquesCarette** [comment]
So that last part probably is best done by me - and thus the correct assignees.
